src/components/TodosProvider/TodosProvider.tsx

```diff
@@ -0,0 +1,233 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from '../../api/todos';
+
+import { ErrorMessage } from '../../types/ErrorMessage';
+import { FilterTypes } from '../../types/FIlterTypes';
+import { Todo, UpdateData } from '../../types/Todo';
+import { getFilteredTodos } from '../../utils/getFilteredTodos';
+
+const USER_ID = 6359;
+
+export type ContextType = {
+  todos: Todo[],
+  tempTodo: Todo | null,
+  processedTodos: Todo[],
+  activeTodos: Todo[],
+  completedTodos: Todo[],
+  hasError: boolean,
+  errorMessage: ErrorMessage,
+  inputDisabled: boolean,
+  handleAddTodo: (todoTitle: string) => void,
+  handleDelete: (todo: Todo) => void,
+  handleUpdateTodo: (todo: Todo, fieldsToUpdate: UpdateData) => void,
+  handleToggleAll: () => void,
+  changeHasError: (value: boolean) => void,
+  deleteAllCompleted: () => void,
+  USER_ID: number,
+};
+
+export const TodosContext = React.createContext<ContextType>({
+  todos: [],
+  tempTodo: null,
+  processedTodos: [],
+  activeTodos: [],
+  completedTodos: [],
+  hasError: false,
+  errorMessage: ErrorMessage.NONE,
+  inputDisabled: false,
+  handleAddTodo: () => { },
+  handleDelete: () => { },
+  handleUpdateTodo: () => { },
+  handleToggleAll: () => { },
+  changeHasError: () => { },
+  deleteAllCompleted: () => { },
+  USER_ID,
+});
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const TodosProvider: React.FC<any> = (
+  { children },
+) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processedTodos, setProcessedTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState<boolean>(false);
+  const [
+    errorMessage,
+    setErrorMessage,
+  ] = useState<ErrorMessage>(ErrorMessage.NONE);
+  const [inputDisabled, setInputDisabled] = useState<boolean>(false);
+
+  const fetchUserTodos = useCallback(async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setErrorMessage(ErrorMessage.FIND);
+      setHasError(true);
+    }
+  }, []);
+
+  useEffect(() => {
+    fetchUserTodos();
+  }, []);
+
+  const handleAddTodo = useCallback(async (todoTitle: string) => {
+    if (!todoTitle) {
+      setErrorMessage(ErrorMessage.EMPTY_TITLE);
+      setHasError(true);
+
+      return;
+    }
+
+    const newTodo = {
+      id: 0,
```

Comment:
it is not a good idea, to send todo with `0` id, every todo will have 0 id (which should be unique). It is not necessary to pass id to api, it will be automatically calculated on backend